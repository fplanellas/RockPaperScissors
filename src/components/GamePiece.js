import React from 'react';
import styled from 'styled-components';

const GamePieceStyled = styled.div`
    width: 100px;
    height: 100px;
    border: 5px solid #9b9b9b;
    border-radius: 50%;
    display: flex; 
    justify-content: center;
    align-items: center;
    background: silver;
    cursor: pointer;
`

const GamePiece = ({ name = '', clicked }) => {
 
    const handleClick = () => {
        if (clicked) {
            clicked(name);
        };      
    };

  return (
    <GamePieceStyled onClick={ handleClick }>
      <img src={`./assets/${name}.png`} alt="" width="80px"/>
    </GamePieceStyled>
  )
}

export default GamePiece
