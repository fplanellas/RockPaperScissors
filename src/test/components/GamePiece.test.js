import React from 'react';
import { shallow } from 'enzyme';
import GamePiece from '../../components/GamePiece';

describe('testing in main component GamePiece', () => {

    test('should be displayed correctly', () => {
        const wrapper = shallow( <GamePiece /> );
        expect( wrapper ).toMatchSnapshot();
    })
      
})