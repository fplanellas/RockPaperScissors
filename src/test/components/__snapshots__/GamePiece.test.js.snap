// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing in main component GamePiece should be displayed correctly 1`] = `
<styled.div
  onClick={[Function]}
>
  <img
    alt=""
    src="./assets/.png"
    width="80px"
  />
</styled.div>
`;
